rsconf/package_data/rsconf/rsconf.sh

```diff
@@ -27,32 +27,42 @@ rsconf_append() {
 }
 
 rsconf_append_authorized_key() {
-    local user=$1
-    local key=$2
-    local ssh_d="$( getent passwd "$user" | cut -d: -f6 )"/.ssh
+    declare user=$1
+    declare key=$2
+    declare ssh_d="$( getent passwd "$user" | cut -d: -f6 )"/.ssh
     if [[ ! -e "$ssh_d" ]]; then
         install -d -m 700 -o "$user" -g "$user" "$ssh_d"
     fi
-    local keys_f=$ssh_d/authorized_keys
+    declare keys_f=$ssh_d/authorized_keys
     if [[ ! -e $keys_f ]]; then
         install -m 600 -o "$user" -g "$user" /dev/null "$keys_f"
     fi
     rsconf_edit_no_change_res=0 rsconf_append "$keys_f" "$key"
 }
 
+rsconf_clone_repo() {
+    declare repo=$1
+    declare dest=$2
+    declare user=$3
+    if [[ ! -d $dest || ! $(ls -A "$dest") ]]; then
```

Comment:
Why the need for the second check? If it is a directory, it should not be cloned.

Reply:
We need to be able to handle the fact that `devbox_<user>.sh` is going to create the full directory structure. Ex
```
rsconf_install_access '700' 'vagrant' 'vagrant'
rsconf_install_directory '/srv/devbox_xxx/jupyter'
rsconf_install_directory '/srv/devbox_xxx/src'
rsconf_install_directory '/srv/devbox_xxx/src/biviosoftware'
rsconf_install_directory '/srv/devbox_xxx/src/biviosoftware/home-env'
rsconf_clone_repo https://github.com/biviosoftware/home-env.git /srv/devbox_xxx/src/biviosoftware/home-env vagrant
```

Reply:
OK.